File: onion-architecture/java/src/main/java/com/example/infrastructure/adapter/StudentRepositoryImpl.java
package com.example.infrastructure.adapter;

import com.example.application.ports.StudentRepository;
import com.example.domain.Student;
import org.springframework.stereotype.Repository;
import java.util.*;

@Repository
public class StudentRepositoryImpl implements StudentRepository {

    private final Map<Long, Student> store = new HashMap<>();

    @Override
    public Optional<Student> findById(Long id) {
        return Optional.ofNullable(store.get(id));
    }

    @Override
    public void save(Student student) {
        store.put(student.getId(), student);
    }
}

File: onion-architecture/kotlin/src/main/kotlin/com/example/infrastructure/adapter/StudentRepositoryImpl.kt
package com.example.infrastructure.adapter

import com.example.application.ports.StudentRepository
import com.example.domain.Student
import org.springframework.stereotype.Repository

@Repository
class StudentRepositoryImpl : StudentRepository {

    private val store: MutableMap<Long, Student> = mutableMapOf()

    override fun findById(id: Long): Student? {
        return store[id]
    }

    override fun save(student: Student) {
        store[student.id] = student
    }
}